src/main/java/org/folio/linked/data/service/impl/ResourceServiceImpl.java

```diff
@@ -257,4 +259,23 @@ private void breakEdgesAndDelete(Resource resource) {
     breakCircularEdges(resource);
     resourceRepo.delete(resource);
   }
+
+  private void refreshWork(Resource resource) {
+    if (resource.isOfType(INSTANCE)) {
+      extractWorkFromInstance(resource)
+        .ifPresent(work -> {
+          edgeRepo.findByIdTargetHash(work.getId())
+            .forEach(work::addIncomingEdge);
+          addOutgoingEdges(work);
+        });
+    }
+  }
+
+  private void addOutgoingEdges(Resource resource) {
+    edgeRepo.findByIdSourceHash(resource.getId())
+      .forEach(resourceEdge -> {
+        resource.addOutgoingEdge(resourceEdge);
+        addOutgoingEdges(resourceEdge.getTarget());
```

Comment:
This is dangerous right? It will result in an infinite navigation of all nodes in the graph.

Reply:
I've added the ```edgeDepth``` parameter to avoid such problem (just like in ```ResourceTestService``` class)